app/models/concerns/visible.rb

```diff
@@ -1,19 +1,11 @@
+# frozen_string_literal: true
+
 module Visible
   extend ActiveSupport::Concern
 
-  VALID_STATUSES = %w[public private archived]
-
-  included do
-    validates :status, inclusion: { in: VALID_STATUSES }
-  end
-
   class_methods do
     def public_count
-      where(status: 'public').count
+      where(status: 1).count
```

Comment:
It's better not to use "magic numbers". They are not informative.
You can do something like `self.statuses[:public]` 
Not 100% sure it will work, check, please. If I understand correctly, `self` on the level where you call `public_count` should be a your model class.